core/utils.ts

```diff
@@ -245,6 +245,110 @@ export function validateQueryString(session: Session, query: string, filename: s
   }
 }
 
+/**
+ * Checks if the Cloud Resource connection has a valid format.
+ * @param session Used to throw a compile error if connection is not valid.
+ * @param connection String to be validated.
+ * @param filename Used to throw a compile error if connection is not valid.
+ */
+export function validateConnectionFormat(
```

Comment:
New functions here require some unit testing especially because we are relying on magic regexes. Eventually I plan to cover all utils with unit tests.